### PR TITLE
Anti-ameridian weaponry

### PR DESCRIPTION
**Pull request by r4d6**
## About The Pull Request
1. Add the handheld sonic emitter, an experimental weapon designed to destroy ameridian-based lifeform by targeting the resonant frequency of the crystal.
.
In game term, it is a gun that deal 10 brute damage, with an armor penetration of 90, and does 20x damage against ameridian golems, for a total theoretical damage per shot of 200 against them. However, golems killed this way do not drop any loot at all.
.
.
2. Add the sonic fence generator, a type of shield generator that, while not blocking anything, destroy ameridian crystals that cross it by targeting its resonant frequency, including golems. However, any disruption to that frequency, say by the crystal being in a bag or in a hand, is enough to let the crystal survive the passage. It also prevent crystals from spreading as long as the shield is active.
.
In game term, the sonic fence doesn't stop anything, but destroy ameridian golems that cross it, also destroying any loot. Any shard dragged through the fence is likewise destroyed, though shards carried by a person are left intact. The crystal cannot spread to a tile occupied by the actual fence, though a spire can spread past it if the fence is too tight.

Those two additions are designed to help in containing and maintaining containment of ameridium at the expense of any potential loot.

The content of this PR will **not** be introduced in next Saturday's event. Instead the availability of the content included in this PR will depend on the colony's performance.

This PR is in Draft mode to get people's opinion on it and until the sprites for the gun is done.

### CODE REVIEW

File: code/game/objects/items/weapons/circuitboards/machinery/ameridian.dm [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+	board_type = "machine"
+	origin_tech = list(TECH_DATA = 4, TECH_MATERIAL = 9, TECH_ENGINEERING = 5)
+	req_components = list(
+		/obj/item/stock_parts/matter_bin = 1,
+		/obj/item/stock_parts/manipulator = 1
+	)
+
+/obj/item/circuitboard/ameridianshieldwallgen
+	name = "circuit board ameridian sonic fence generator"
+	build_name = ""
```
Comment (Trilbyspaceclone) [inline code/game/objects/items/weapons/circuitboards/machinery/ameridian.dm:13]: did you mean to do this?

File: code/modules/ameridian/ameridian_crystal.dm [hunk 2]
```diff
@@ -89,17 +96,30 @@
...
 	if(turf_list.len)
 		var/turf/T = pick(turf_list)
 		new /obj/structure/ameridian_crystal(T) // We spread
 
+// Check the given turf to see if there is any special things that would prevent the spread
+/obj/structure/ameridian_crystal/proc/can_spread_to_turf(var/turf/T)
+	if(T)
+		if(istype(T, /turf/space)) // We can't spread in SPACE!
+			return FALSE
+		if(istype(T, /turf/simulated/open)) // Crystals can't float. Yet.
```
Comment (Trilbyspaceclone) [inline code/modules/ameridian/ameridian_crystal.dm:113]: You sure they dont just spread then fall down a level?
If it works like hivemind does that would be really cool to have a multy layered deep underground were shards fall form above

**Comment by Trilbyspaceclone** [comment]
Really loving the shard work and updates with it!
<3

File: code/game/objects/items/weapons/circuitboards/machinery/ameridian.dm (hunk 1 above)
Comment (r4d6) [reply code/game/objects/items/weapons/circuitboards/machinery/ameridian.dm:13]: I'll fix it

**Comment by r4d6** [comment]
Sprite for the gun isn't done yet, but I don't plan on introducing it in the next event, so it is fine.

**Pull request merged**